src/test/java/org/apache/sysds/test/functions/mlcontext/MLContextPNMFTest.java

```diff
@@ -48,17 +49,26 @@
 	private final static double maxiter = 10;
 
 	@Test
-	public void testPNMFSparse() {
-		runPNMFTestMLC(true);
+	public void testPNMFSparseCP() {
+		runPNMFTestMLC(true, LopProperties.ExecType.CP);
 	}
 
 	@Test
-	public void testPNMFDense() {
-		runPNMFTestMLC(false);
+	public void testPNMFSparseSP() {
+		runPNMFTestMLC(true, LopProperties.ExecType.SPARK);
 	}
 
+	@Test
+	public void testPNMFDenseCP() {
+		runPNMFTestMLC(false, LopProperties.ExecType.CP);
+	}
+
+	@Test
+	public void testPNMFDenseSP() {
+		runPNMFTestMLC(false, LopProperties.ExecType.SPARK);
+	}
 
-	private void runPNMFTestMLC(boolean sparse) {
+	private void runPNMFTestMLC(boolean sparse, LopProperties.ExecType instType) {
```

Comment:
See [AlgorithmPCA](https://github.com/apache/systemds/blob/master/src/test/java/org/apache/sysds/test/functions/codegenalg/partone/AlgorithmPCA.java)

> 	private final static int cols1 = 1007;
	private final static int cols2 = 387;

[#L158](https://github.com/apache/systemds/blob/8be71ec01a80fa98b6e0e088a4476f65aabe7707/src/test/java/org/apache/sysds/test/functions/codegenalg/partone/AlgorithmPCA.java#L158)
> 			int cols = (instType==ExecType.SPARK) ? cols2 : cols1;


Tip: When the input dimension is more the code will be executed in `spark` instead of `CP`.

So, during `testPNMFDenseSP()` have more number of columns!

Reply:
if instType is 'spark', then cols = cols2 where, cols2<cols1, so am I interpreting it wrong ways with correspondence to the tip?
And regarding testPNMF, col is initialized to a fixed value, so am I supposed to create a new variable?

Reply:
Yes.

Reply:
As a check when the program run by spark in the `HeavyHitterInstructions` one would see the keyword `spark`! 😺

Reply:
![image](https://user-images.githubusercontent.com/62425457/96143361-a665ed80-0f20-11eb-95f7-51a6706af82d.png)